test/browser/fake_events.js

```diff
@@ -2,51 +2,57 @@
  * Helper function used in browser tests to simulate HTML5 events
  */
 
-function simulateKeyEvent(eventType, keyCode, code) {
+function simulateKeyEvent(target, eventType, keyCode, code) {
   var props = { keyCode, charCode: keyCode, view: window, bubbles: true, cancelable: true };
   if (code) props['code'] = code;
   var event = new KeyboardEvent(eventType, props);
-  return document.dispatchEvent(event);
+  return target.dispatchEvent(event);
 }
 
-function simulateKeyDown(keyCode, code = undefined) {
-  var doDefault = simulateKeyEvent('keydown', keyCode, code);
+function simulateKeyDown(target, keyCode, code = undefined) {
+  var doDefault = simulateKeyEvent(target, 'keydown', keyCode, code);
   // As long as not handler called `preventDefault` we also send a keypress
   // event.
   if (doDefault) {
-    simulateKeyEvent('keypress', keyCode, code);
+    simulateKeyEvent(target, 'keypress', keyCode, code);
   }
 }
 
-function simulateKeyUp(keyCode, code = undefined) {
-  simulateKeyEvent('keyup', keyCode, code);
+function simulateKeyUp(target, keyCode, code = undefined) {
+  simulateKeyEvent(target, 'keyup', keyCode, code);
 }
 
-function simulateMouseEvent(x, y, button, absolute) {
+function simulateKeyDownUp(target, keyCode, code = undefined) {
+  simulateKeyEvent(target, 'keyodwn', keyCode, code);
+  simulateKeyEvent(target, 'keyup', keyCode, code);
```

Comment:
Should `simulateKeyDownUp` perhaps call `simulateKeyDown, simulateKeyUp`? That would reuse more code and avoid hardcoding the event names here.

Reply:
Oops, yes of course.